arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -135,9 +135,13 @@ void DatabaseManagerThread::run() {
         for (auto it = dropped.begin(), end = dropped.end(); it != end; ++it) {
           TRI_ASSERT(*it != nullptr);
           if ((*it)->isDangling()) {
-            database.reset(*it);
+            // found a database to delete
+            // now move the to-be-deleted database from _droppedDatabases
+            // into the unique_ptr
+            auto* p = *it;
             dropped.erase(it);
-            break;  // found a database to delete
+            database.reset(p);
+            break;
```

Comment:
As we talk, you're right and it's unnecessary, absl flat_hash_map erase no throw if dtor no throw